register-computer/main.py

```diff
@@ -98,6 +102,47 @@ def __connect_to_activedirectory():
 def __generate_password(length=40):
     return str(uuid.uuid4()) + "-" + str(uuid.uuid4())
 
+def __shorten_computer_name(computer_name, gce_instance):    
+    # Look for MIG related metadata
+    has_metadata = ("metadata" in gce_instance.keys() and
+                    "items" in gce_instance["metadata"])
+    if has_metadata:
+      metadata_created_by = next((x for x in gce_instance["metadata"]["items"] if x["key"] == "created-by"), None)
+
+    is_mig = (has_metadata and 
+              metadata_created_by and              
+              "instanceGroupManagers" in metadata_created_by["value"])
+    
+    if not is_mig:
```

Comment:
Consider creating separate methods `__is_managed_instance_group_member` and `__is_gke_nodepool_member` to separate the logic to determine whether it's a MIG/node pool from the name shortening logic

Reply:
created __is_gke_nodepool_member and __get_managed_instance_group_for_instance
for MIG, it's more useful to get back the MIG name, instead of a boolean.
We will use this when we update the function to add MIG instances to an AD group for the MIG